### PR TITLE
refactor(2701): llm prompts to text files

### LINKED ISSUE #2701: refactor LLM prompts
The current prompts are written in `.rs` files. Instead write them into a `.md` file with placeholders using mustache templates `{{}}`.

**Technical Requirements**
- Use `include_str!` to import the prompt.

**Additional Information**
- Most of the logic is in `infer_type_names`

### PR DESCRIPTION
**Pull request by asr2003**
**Summary:**  
Moved prompts to .md file following with placeholders using mustache templates {{}}.

**Issue Reference(s):**  
Fixes #2701 
/claim #2701

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/cli/llm/infer_type_name.rs [hunk 1]
```diff
@@ -55,20 +61,19 @@ impl TryInto<ChatRequest> for Question {
             ],
         })?;
 
+        let template_str = include_str!("prompts.md");
+        let template = mustache::compile_str(template_str).map_err(|e| Error::TemplateError(e.to_string()))?;
```
[the remark this answers is not in the document]
Comment (asr2003) [reply src/cli/llm/infer_type_name.rs]: Just added to reflect any errors :)

File: src/cli/llm/infer_type_name.rs [hunk 2]
```diff
@@ -55,20 +61,19 @@ impl TryInto<ChatRequest> for Question {
             ],
         })?;
 
+        let template_str = include_str!("prompts.md");
+        let template = mustache::compile_str(template_str)?;
+
+        let context = MapBuilder::new()
+            .insert_str("input", &input)
+            .insert_str("output", &output)
+            .build();
```
Comment (tusharmath) [inline src/cli/llm/infer_type_name.rs:70]: can use a maplit macro instead.

File: src/cli/llm/infer_type_name.rs [hunk 3]
```diff
@@ -32,6 +33,11 @@ struct Question {
     fields: Vec<(String, String)>,
 }
 
+struct PromptContext {
+    input: String,
+    output: String,
+}
```
[the remark this answers is not in the document]
Comment (asr2003) [reply src/cli/llm/infer_type_name.rs:39]: Yes, will remove it

**Comment by tusharmath** [comment]
Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

File: Cargo.toml [hunk 4]
```diff
@@ -181,6 +181,7 @@ insta = { workspace = true }
 tempfile = "3.10.1"
 temp-env = "0.3.6"
 maplit = "1.0.2"
+mustache = "0.9.0"
```
Comment (laststylebender14) [inline Cargo.toml:184]: we've built our own mustache, please use that instead of adding new lib.

File: src/cli/llm/infer_type_name.rs [hunk 5]
```diff
@@ -55,20 +57,19 @@ impl TryInto<ChatRequest> for Question {
             ],
         })?;
 
+        let template_str = include_str!("prompts.md");
```
Comment (laststylebender14) [inline src/cli/llm/infer_type_name.rs:60]:
```suggestion
        let template_str = include_str!("prompts/system.md");
```

**Review by tusharmath — APPROVED** [review]

**Pull request merged**